### PR TITLE
Presentation: Ruleset variables docs

### PR DESCRIPTION
**Pull request by grigasp**

### CODE REVIEW

File: full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        }, {
+          ruleType: RuleTypes.RootNodes,
+          condition: `GetVariableStringValue("${variableId}") = "elements"`,
+          specifications: [{
+            specType: ChildNodeSpecificationTypes.InstanceNodesOfSpecificClasses,
+            classes: { schemaName: "BisCore", classNames: ["Element"] },
+            arePolymorphic: true,
+          }],
+        }],
+      };
```
Comment (roluk) [inline full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts:51]: As someone encountering ruleset variables for the first time, I had to stop and think whether "${variableId}" on line 36 is a part of rules variable syntax. Of course, they aren't, but I think the examples would be more obvious if string templating was avoided. Also, I would give this ruleset variable another id: `"var-tree-type"`.

**Review by roluk — CHANGES_REQUESTED** [review]

File: full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts (hunk 1 above)
Comment (grigasp) [reply full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts]: Agree - hardcoded values are more obvious to understand. Note: I didn't prefix variable ID with `var-` - seems unnecessary to me.

File: docs/learning/presentation/RulesetVariables.md [hunk 2]
```diff
@@ -1,86 +1,97 @@
...
-}
+> Ruleset Variables is a new name for now deprecated *User Settings* concept.
+
+Presentation rulesets have a concept called *ruleset variables* whose purpose is to allow modifying ruleset behavior during the session without having to change
+the ruleset itself. The values of ruleset variables can be accessed using [ECExpressions](./ECExpressions.md#ruleset-variables-user-settings) in:
+
+- Rule conditions ([example](#using-variables-in-rule-condition)).
+- Instance filters ([example](#using-variables-in-instance-filter)).
+- Some ECExpressions-based values, like [LabelOverride.label]($presentation-common) ([example](#using-variables-in-ecexpression-based-value)).
+
```
Comment (roluk) [inline docs/learning/presentation/RulesetVariables.md:11]: This phrasing suggests that Ruleset Variables cannot be used in all ECExpressions. Is that the case?

File: docs/learning/presentation/RulesetVariables.md [hunk 3]
```diff
@@ -1,86 +1,97 @@
...
+The advantages of using ruleset variables over values hardcoded into the ruleset are:
+
+- When used for creating a hierarchy, changing a ruleset variable causes only affected parts of the hierarchy to be re-created. Changing a rule in a ruleset causes the whole hierarchy to be re-created.
+- It's much more convenient and effective to pass a large number of IDs through a ruleset variable than it is to hardcode them into a ruleset. Ruleset variables are sent to the backend separately from the ruleset and that allows us to compress them then needed.
+
```
Comment (roluk) [inline docs/learning/presentation/RulesetVariables.md:16]:
```suggestion
- It's much more convenient and effective to pass a large number of IDs through a ruleset variable than it is to hardcode them into a ruleset. Ruleset variables are sent to the backend separately from the ruleset and that allows us to compress them when needed.
```

File: docs/learning/presentation/RulesetVariables.md [hunk 4]
```diff
@@ -1,86 +1,97 @@
...
+Setting the ruleset value to specific IDs creates the hierarchy only for specific elements:
+
+```ts
+[[include:RulesetVariables.InInstanceFilter.SetIds]]
+```
+
+![Filtered Elements' Hierarchy](./media/ruleset-variables/filtered-elements-hierarchy.png)
+
+Clearing the ruleset value brings us back to initial, unfiltered, view:
+
```
Comment (roluk) [inline docs/learning/presentation/RulesetVariables.md:70]:
```suggestion
Clearing the ruleset value brings us back to the initial unfiltered view:
```

**Review by roluk — CHANGES_REQUESTED** [review]

File: full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts (hunk 1 above)
Comment (roluk) [reply full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts]: Thanks. Maybe then call out the name of the variable in the preceding text so the reader knows what to pay attention to?

File: docs/learning/presentation/RulesetVariables.md (hunk 2 above)
Comment (grigasp) [reply docs/learning/presentation/RulesetVariables.md]: Removed `Some`

File: full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts (hunk 1 above)
Comment (grigasp) [reply full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts]: - Made the IDs `UPPER_CASE`, which, I think, exposes them more.
- Added comments next to the ruleset definitions explaining what is controlled by which ruleset variable.

File: docs/learning/presentation/RulesetVariables.md (hunk 2 above)
Comment (roluk) [reply docs/learning/presentation/RulesetVariables.md]: Thanks

File: full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts (hunk 1 above)
Comment (roluk) [reply full-stack-tests/presentation/src/learning-snippets/RulesetVariables.test.ts]: Looks great now, thanks.

**Review by roluk — APPROVED** [review]

**Pull request merged**